src/test/unit/DelegationUnit.t.sol

```diff
@@ -578,6 +602,77 @@ contract DelegationManagerUnitTests_RegisterModifyOperator is DelegationManagerU
         emit AVSMetadataURIUpdated(defaultAVS, metadataURI);
         delegationManager.updateAVSMetadataURI(metadataURI);
     }
+
+    // @notice Verifies an operator registers successfull to avs and see an `OperatorAVSRegistrationStatusUpdated` event emitted
+    function testFuzz_registerOperatorToAVS(bytes32 salt) public {
+        cheats.prank(defaultAVS);
+        cheats.expectEmit(true, true, true, true, address(delegationManager));
+
+        address operator = cheats.addr(delegationSignerPrivateKey);
+        emit OperatorAVSRegistrationStatusUpdated(operator, defaultAVS, OperatorAVSRegistrationStatus.REGISTERED);
+
+        uint256 expiry = type(uint256).max;
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature = _getOperatorSignature(
+            delegationSignerPrivateKey,
+            operator,
+            defaultAVS,
+            salt,
+            expiry
+        );
+
+        delegationManager.registerOperatorToAVS(operator, operatorSignature);
+        cheats.stopPrank();
+    }
+
+    // @notice Verifies an operator registers fails when the signature is not from the operator
+    function testFuzz_revert_whenSignatureAddressIsNotOperator(bytes32 salt) public {
+        cheats.prank(defaultAVS);
+        address operator = cheats.addr(delegationSignerPrivateKey);
+
+        uint256 expiry = type(uint256).max;
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature = _getOperatorSignature(
+            delegationSignerPrivateKey,
+            operator,
+            defaultAVS,
+            salt,
+            expiry
+        );
+
+        cheats.prank(operator);
+        cheats.expectRevert("EIP1271SignatureUtils.checkSignature_EIP1271: signature not from signer");
+        delegationManager.registerOperatorToAVS(operator, operatorSignature);
+        cheats.stopPrank();
+    }
+
+    // @notice Verifies an operator registers fails when the signature expiry already expires
+    function testFuzz_revert_whenExpiryHasExpired(bytes32 salt, uint256 expiry, ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature) public {
+        cheats.prank(defaultAVS);
+        address operator = cheats.addr(delegationSignerPrivateKey);
+        cheats.assume(operatorSignature.expiry < block.timestamp);
+
+        cheats.expectRevert("DelegationManager.registerOperatorToAVS: operator signature expired");
+        delegationManager.registerOperatorToAVS(operator, operatorSignature);
+    }
+
+    // @notice Verifies an operator registers fails when it's already registered to the avs
+    function testFuzz_revert_whenOperatorAlreadyRegisteredToAVS(bytes32 salt) public {
+        cheats.prank(defaultAVS);
```

Comment:
Nit: Move prank to be right above `registerOperatorToAVS` and use `cheats.startPrank(defaultAVS)` since we are pranking over multiple calls. Reason why is passes regardless is that the defaultAVS is the contract address. Technically pranking isn't needed but I like it as it is clear to the reader of tests what's going on.